sycl/doc/LinkedAllocations.md

```diff
@@ -0,0 +1,74 @@
+## Brief overview of allocations for memory objects
+
+A SYCL memory object (`buffer`/`image`) can be accessed in multiple contexts
+throughout its lifetime. Since this is dynamic information that is unknown
+during memory object construction, no allocation takes place at that point.
+Instead, memory is allocated in each context whenever the SYCL memory object
+is first accessed there:
+
+```C++
```

Comment:
Also current version of Sphinx doesn't process these hints correctly (github also has hard times rendering my suggestion correctly, but I propose to remove `C++`).
```suggestion
```<just_quotes>
```